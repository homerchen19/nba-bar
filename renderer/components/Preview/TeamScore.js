import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '../../styles/theme';
import nba from '../../utils/nba';

const Wrapper = styled(Flex)`
  flex: 1;
  width: 100%;
  color: ${colors.white};
`;

const TeamWrapper = styled(Flex)`
  flex: 3;
  width: 100%;
  padding: 3px 0;
  border-top: 1px solid ${colors.white};
  background: linear-gradient(
    to right,
    ${props => props.background.home} 50%,
    ${props => props.background.visitor} 50%
  );
`;

const InfoWrapper = styled(Flex)`
  flex: 2;
  width: 100%;
  padding: 0;
  background: ${colors.darkBlue};
`;

const TeamName = styled.h3`
  line-height: 25px;
`;

const Arena = styled.p`
  margin-top: 4px;
  font-size: 9px;
`;

const Team = ({ team }) => (
  <Flex direction="column">
    <TeamName>{team.name}</TeamName>
    <p>{`${team.w} - ${team.l}`}</p>
  </Flex>
);

const TeamScore = ({ time, arena, city, home, visitor }) => (
  <Fragment>
    <Wrapper justify="start" direction="column">
      <InfoWrapper direction="column">
        <h4>{time}</h4>
        <Arena>{`${arena}, ${city}`}</Arena>
      </InfoWrapper>
      <TeamWrapper
        justify="center"
        background={{
          home: nba.getTeamBackgroundColor(home.name),
          visitor: nba.getTeamBackgroundColor(visitor.name),
        }}
      >
        <Flex.Item>
          <Team team={home} />
        </Flex.Item>
        <Flex.Item style={{ margin: 0 }}>
          <Team team={visitor} />
        </Flex.Item>
      </TeamWrapper>
    </Wrapper>
  </Fragment>
);

Team.propTypes = {
  team: PropTypes.object.isRequired,
};

TeamScore.propTypes = {
  time: PropTypes.string.isRequired,
  arena: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default TeamScore;
