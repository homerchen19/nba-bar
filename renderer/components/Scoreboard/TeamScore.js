import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '@styles/theme';
import nba from '@utils/nba';
import { BarLoader } from '../Loader';

const Wrapper = styled(Flex)`
  width: 100%;
  color: #fff;
`;

const InfoWrapper = styled(Flex)`
  width: 100%;
  padding: 0 0 5px;
  background: ${colors.darkBlue};
`;

const ScoreWrapper = styled(Flex)`
  width: 100%;
  padding: 3px 0 6px;
  border-top: 1px solid ${colors.white};
  background: linear-gradient(
    to right,
    ${props => props.background.home} 50%,
    ${props => props.background.visitor} 50%
  );
`;

const Team = styled(Flex)`
  flex: 1;
  margin: 0;
`;

const TeamName = styled.h3`
  line-height: 35px;
`;

const Score = styled.p`
  opacity: 1;
  font-size: large;
  font-weight: ${props => (props.win ? '800' : '300')};
  text-align: center;
`;

const TeamScore = ({
  arena,
  city,
  home,
  visitor,
  winner,
  gameStatus,
  showBarLoader,
}) => (
  <Wrapper justify="start" direction="column">
    <InfoWrapper direction="column" justify="center" align="center">
      <Flex.Item style={{ textAlign: 'center' }}>
        <h4>{gameStatus}</h4>
      </Flex.Item>
      <Flex.Item style={{ margin: 0, textAlign: 'center' }}>
        <span style={{ fontSize: '10px' }}>{`${arena}, ${city}`}</span>
      </Flex.Item>
    </InfoWrapper>
    {showBarLoader && <BarLoader />}
    <ScoreWrapper
      justify="center"
      align="center"
      background={{
        home: nba.getTeamBackgroundColor(home.name),
        visitor: nba.getTeamBackgroundColor(visitor.name),
      }}
    >
      <Team direction="column" justify="center" align="center">
        <TeamName>{home.name}</TeamName>
        <Score win={winner === 'home'}>{home.score}</Score>
      </Team>
      <Team direction="column" justify="center" align="center">
        <TeamName>{visitor.name}</TeamName>
        <Score win={winner === 'visitor'}>{visitor.score}</Score>
      </Team>
    </ScoreWrapper>
  </Wrapper>
);

TeamScore.propTypes = {
  arena: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  winner: PropTypes.string.isRequired,
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
  gameStatus: PropTypes.string.isRequired,
  showBarLoader: PropTypes.bool,
};

TeamScore.defaultProps = {
  showBarLoader: false,
};

export default TeamScore;
