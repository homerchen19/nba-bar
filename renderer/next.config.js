const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
