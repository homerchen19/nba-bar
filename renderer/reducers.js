import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
