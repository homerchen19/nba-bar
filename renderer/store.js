import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
