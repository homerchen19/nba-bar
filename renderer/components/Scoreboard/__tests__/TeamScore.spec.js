import React from 'react';
import renderer from 'react-test-renderer';

import TeamScore from '../TeamScore';

it('be defined', () => {
  expect(TeamScore).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    arena: 'Bankers Life Fieldhouse',
    city: 'Indianapolis',
    gameStatus: 'Final',
    home: {
      name: 'IND',
      score: '126',
    },
    visitor: {
      name: 'GSW',
      score: '106',
    },
    winner: 'home',
  };
  const tree = renderer.create(<TeamScore {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
