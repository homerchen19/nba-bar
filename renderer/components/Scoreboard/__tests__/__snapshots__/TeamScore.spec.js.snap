// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox TeamScore__Wrapper-dpbnwk-0 fPRylN am-flexbox-dir-column am-flexbox-justify-start am-flexbox-align-center"
>
  <div
    className="am-flexbox TeamScore__InfoWrapper-dpbnwk-1 cgGrQF am-flexbox-dir-column am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      className="am-flexbox-item"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h4>
        Final
      </h4>
    </div>
    <div
      className="am-flexbox-item"
      style={
        Object {
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "fontSize": "10px",
          }
        }
      >
        Bankers Life Fieldhouse, Indianapolis
      </span>
    </div>
  </div>
  <div
    background={
      Object {
        "home": "#041e42",
        "visitor": "#ffc72d",
      }
    }
    className="am-flexbox TeamScore__ScoreWrapper-dpbnwk-2 jMzxzS am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      className="am-flexbox TeamScore__Team-dpbnwk-3 bBlydO am-flexbox-dir-column am-flexbox-justify-center am-flexbox-align-center"
    >
      <h3
        className="TeamScore__TeamName-dpbnwk-4 kDLhxp"
      >
        IND
      </h3>
      <p
        className="TeamScore__Score-dpbnwk-5 bsFBtt"
      >
        126
      </p>
    </div>
    <div
      className="am-flexbox TeamScore__Team-dpbnwk-3 bBlydO am-flexbox-dir-column am-flexbox-justify-center am-flexbox-align-center"
    >
      <h3
        className="TeamScore__TeamName-dpbnwk-4 kDLhxp"
      >
        GSW
      </h3>
      <p
        className="TeamScore__Score-dpbnwk-5 fmmlWw"
      >
        106
      </p>
    </div>
  </div>
</div>
`;
